porch/pkg/registry/porch/packagecommon.go

```diff
@@ -141,30 +145,71 @@ func (r *packageCommon) updatePackageRevision(ctx context.Context, name string,
 
 	oldPackage, err := r.getPackage(ctx, name)
 	if err != nil {
-		return nil, false, err
+		if apierrors.IsNotFound(err) {
+			// For server-side apply, we create new objects where they don't already exist
+		} else {
+			return nil, false, err
+		}
+	}
+
+	isCreate := false
```

Comment:
[nit] I found this line surprising given 149 determined that we may proceed to creation.

I'd probably go for:

var isCreate bool
if ... {
  isCreate = true
} else {
}

or add a comment, or move the definition above line 146

Reply:
Good point; reorganized the code to clean this up